libs/ledger-live-common/src/ble/hooks/useBleDevicesScanning.ts

```diff
@@ -0,0 +1,170 @@
+import { useState, useEffect, useRef, useCallback } from "react";
+import { Observable, of } from "rxjs";
+import { concatAll } from "rxjs/operators";
+import { getInfosForServiceUuid } from "@ledgerhq/devices";
+import type { DeviceModel, DeviceModelId } from "@ledgerhq/devices";
+import type {
+  Observer as TransportObserver,
+  Subscription as TransportSubscription,
+  DescriptorEvent,
+  DescriptorEventType,
+} from "@ledgerhq/hw-transport";
+
+// Should be exported from somewhere else
+export type TransportBleDevice = {
+  // Device identifier: MAC address on Android and UUID on iOS.
+  id: string;
+  // Device name
+  name: string | null;
+  // User friendly name of device.
+  localName: string | null;
+  // Current Received Signal Strength Indication of device
+  rssi: number | null;
+  // Current Maximum Transmission Unit for this device.
+  // When device is not connected default value of 23 is used.
+  mtu: number;
+  // List of available services visible during scanning.
+  serviceUUIDs: string[] | null;
+  // Allows any other properties
+  [otherOptions: string]: unknown;
+};
+
+export type ScannedDevice = {
+  deviceId: string;
+  deviceName: string;
+  bleRssi: number | null;
+  deviceModel: DeviceModel;
+};
+
+export type UseBleDevicesScanningResult = {
+  scannedDevices: ScannedDevice[];
+  scanningTimedOut: boolean;
+  scanningError: Error | null;
+};
+
+export type UseBleDevicesScanningDependencies = {
+  bleTransportListen: (
```

Comment:
dependency injection so we don't depend on the mobile or desktop implementation ;)